### ISSUE TITLE
Changing rancher server default ssl settings to Lets encrypt+certmanager results in Unable to connect to the server: x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by shmulikah** [rancher/rancher#23146]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Install HA rancher 2.2.3 via rke and helm with the default ssl option (rancher issuer and cattle-ca) and replace it with lets encrypt production cluster issuer + cert manager.

**Result:**
Rancher UI can be browsed securely with a valid lets encrypt certificate, but:
- When copying the kubeconfig via UI and trying to run kubectl i'm getting 
`Unable to connect to the server: x509: certificate signed by unknown authority.`
- Same x509 error occurs when i use rancher cli (calling the kubectl command of it)
- When clicking launch kubectl from the UI it opens up the shell and i'm able to run kubectl commands without the error

**Other details that may be helpful:**
- when removing certificate-authority-data from kubeconfig then kubectl works as expected
- i've changed the content of the certificate resource that is associated with the rancher server ingress from cert.pem to the fullchain.pem of the lets encrypt certificate but it didn't help
- I've found that the certifica-authority-data shown in the kubeconfig in the rancher UI is still the old cattle-ca rancher cert that i've replaced.
- When accessing https://<FQDN>/v3/settings/cacerts i can see that the old cattle-ca still there
-
**Environment information**
- Rancher version: `rancher/rancher:v2.2.3`
- Installation option : `HA`

**Cluster information**
- Cluster type: Custom
- Machine type: metal and specifications (CPU/memory) are 16GB ram and 8 cores
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.7", GitCommit:"6f482974b76db3f1e0f5d24605a9d1d38fad9a2b", GitTreeState:"clean", BuildDate:"2019-03-25T02:52:13Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):
```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:38:45 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:38:45 2017
 OS/Arch:      linux/amd64```

**Comment by fizzyduck** [comment]
@shmulikah I've had this same issue with rancher v2.3.0-rc9 via the rancher cli:

When using the rancher CLI tool eg. "rancher kubectl ..." a temporary kubeconfig file is created which  includes the certificate-authority-data, presumably retrieved from the /v3/settings/cacerts endpoint. This results in the same "Unable to connect to the server: x509: certificate signed by unknown authority" error.

Replacing the "certificate-authority-data" with a base64 encoded representation of the certificate retrieved using openssl:

`echo | openssl s_client -servername servername -connect server:443 |  sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p'`

temporarily resolves the problem, but due to the way the rancher command works (the temporary file is transient and thrown away when the command exits) this is not a valid workaround.

**Comment by fizzyduck** [comment]
@shmulikah I think that this may be resolved by #14731

**Comment by superseb** [comment]
Merging in https://github.com/rancher/rancher/issues/14731
